### PR TITLE
test: add basic tests for the ForgotPassword page

### PR DESCRIPTION
**Pull request by icatalina**
## What/Why?
This commit sets the basis for testing across the app.

Adds several testing tools used to ensure the app works correctly.

## Rollout/Rollback
Revert this PR

## Testing
This PR is mostly about unit testing, the build should prove everything works.

### CODE REVIEW

File: apps/storefront/src/shared/service/request/fetch.ts [hunk 1]
```diff
@@ -9,7 +7,7 @@ const responseResult = (res: Response, resolve: (val?: unknown) => void, init: R
 
 function b3Fetch(path: string, init: RequestInit): any {
   return new Promise((resolve, reject) => {
-    originFetch(path, init)
+    fetch(path, init)
```
Comment (icatalina) [inline apps/storefront/src/shared/service/request/fetch.ts:10]: We need to use straight native fetch in order for MSW to work correctly

File: apps/storefront/src/components/RegisteredCloseButton.tsx [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import { Dispatch, SetStateAction, useContext } from 'react';
 import { useNavigate } from 'react-router-dom';
+import { useB3Lang } from '@b3/lang';
```
Comment (bc-marco) [inline apps/storefront/src/components/RegisteredCloseButton.tsx:3]: This import is not necessary

**Review by bc-marco — CHANGES_REQUESTED** [review]
Overall looks good to me
Great test btw, just a couple of errors they seem to be related to fetch operations

image

File: apps/storefront/src/components/RegisteredCloseButton.tsx (hunk 2 above)
Comment (icatalina) [reply apps/storefront/src/components/RegisteredCloseButton.tsx:3]: noticed this, but was afk, will fix in Monday

Comment (icatalina) [reply apps/storefront/src/components/RegisteredCloseButton.tsx]: fixed

**Comment by icatalina** [comment]
Uh, this is interesting, they all pass fine on local:
image

I'll investigate a bit further

**Comment by deov31** [comment]
I'm also seeing those errors while running locally the tests 👀 
image

**Comment by icatalina** [comment]
sorted, was missing a few env. variables in `vite.config.ts`

**Review by bc-marco — APPROVED** [review]

**Review by deov31 — APPROVED** [review]

**Pull request merged**